ingestion/ingestion/__main__.py

```diff
@@ -2,96 +2,127 @@
 
 import logging
 import sys
-from pathlib import Path
+from argparse import ArgumentDefaultsHelpFormatter, ArgumentParser
 
-from pydantic import ValidationError
-
-from ingestion.errors import DataValidationError, ErrorWriter
-from ingestion.filesystem import find_data_files, load_data
-from ingestion.schema import Data
+from ingestion.extract import add_flags as add_extract_flags
+from ingestion.extract import extract_cmd
+from ingestion.ingest import add_add_dataset_flags as add_ingest_add_dataset_flags
+from ingestion.ingest import add_flags as add_ingest_flags
+from ingestion.ingest import ingest_cmd
+from ingestion.logging import setup_logger
 
 logger = logging.getLogger(__name__)
 
 
-def main():
-    import argparse
-    import os
+def split_argv(argv: list[str], delimiter: str) -> list[list[str]]:
+    out: list[list[str]] = []
+    temp: list[str] = []
 
-    parser = argparse.ArgumentParser(
-        description="This is a python script to read c-elegans ingestion"
-        "files and validate its content."
-    )
+    for arg in argv:
+        if arg == delimiter:
+            out.append(temp)
+            temp = [arg]
+            continue
+        temp.append(arg)
+
+    if temp:
+        out.append(temp)
 
-    def directory(raw_path: str) -> Path:
-        if not os.path.isdir(raw_path):
-            raise argparse.ArgumentTypeError(f"{raw_path} is not an existing directory")
-        return Path(os.path.abspath(raw_path))
-
-    parser.add_argument(
-        "-i",
-        "--ingestion-dir",
-        help="input files to be ingested (default: current directory)",
-        type=directory,
-        default=os.path.curdir,
+    return out
+
+
+def _main(argv: list[str] | None = None):
```

Comment:
I initially used `Sequence` instead of `list` because `ArgumentParser` can take `list` or `tuples`, so it seemed like a nice middle ground